delivery/media/watermarking/lib/media-delivery-watermarking.js

```diff
@@ -0,0 +1,1948 @@
+        if (void 0 === payload.wmidtyp) throw new Error("Token malformed: missing mandatory wmidtyp claim");
+        if (0 === payload.wmidtyp) {
+            if (!payload.wmidalg) throw new Error("Token malformed: missing wmidalg claim for direct case (i.e wmidtyp = 0)");
+            if (![ "aes-128-cbc", "aes-256-cbc" ].includes(payload.wmidalg)) throw new Error("Token malformed: invalid wmidalg, must be any of aes-128-cb or aes-256-cbc");
+            if (void 0 === payload.wmidivlen) throw new Error("Token malformed: missing wmidivlen claim for direct case (i.e wmidtyp = 0)");
+            if ("number" != typeof payload.wmidivlen) throw new Error("Token malformed: wmidivlen must be number");
+            if (!payload.wmidivhex) throw new Error("Token malformed: missing wmidivhex claim for direct case (i.e wmidtyp = 0)");
+            if ("string" != typeof payload.wmidivhex) throw new Error("Token malformed: wmidivhex must be string");
+            if (!payload.wmidpid) throw new Error("Token malformed: missing wmidpid claim for direct case (i.e wmidtyp = 0)");
+            if ("string" != typeof payload.wmidpid) throw new Error("Token malformed: wmidpid must be string");
+            if (!payload.wmidpalg) throw new Error("Token malformed: missing wmidpalg claim for direct case (i.e wmidtyp = 0)");
+            if ("sha256" !== payload.wmidpalg) throw new Error("Token malformed: invalid wmidpalg, must be sha256");
+        } else {
+            if (1 !== payload.wmidtyp) throw new Error("Token malformed: invalid wmidtyp, must be 0 or 1");
+            if (payload.wmidkeyver && "number" != typeof payload.wmidkeyver) throw new Error("Token malformed: wmidkeyver must be number");
+            if (void 0 === payload.wmopid) throw new Error("Token malformed: missing wmopid claim for indirect case (i.e wmidtyp = 1)");
+            if (payload.wmopid && ("number" != typeof payload.wmopid || Math.floor(payload.wmopid) !== payload.wmopid)) throw new Error("Token malformed: wmopid must be number");
+        }
+    }
+    async getWMPathWithVariant(path, payload, secretKey, variantSubPath, rangeHeader) {
+        if ("string" != typeof path) throw new Error("Invalid path type, expected string!");
+        if ("object" != typeof payload) throw new Error("Invalid payload type, expected JSON object!");
+        if ("string" != typeof secretKey) throw new Error("Invalid secretKey type, expected hex encoded string!");
+        if (!Array.isArray(variantSubPath)) throw new Error("Invalid variantSubPath type, expected list of object containing field variant and subPath. (e.g [{variant: 0, subPath: A}])!");
+        if (rangeHeader && "string" != typeof rangeHeader) throw new Error("Invalid rangeHeader type, expected string!");
+        if (0 === payload.wmidtyp) throw new Error("Direct case is not supported at the moment!");
+        if (1 === payload.wmidtyp) {
+            if ("irdeto" === payload.wmvnd) {
```

Comment:
Is it possible to generalize this, rather than hardcoding the vendor? The vendor lookup around line 1759 looks like we would only ever get `irdeto` due to this check.

Reply:
done, removed wmvnd specfic check.